app/actions/ControlActions.js

```diff
@@ -1067,35 +1082,144 @@ export const LOCKACCOUNT_SUCCESS = "LOCKACCOUNT_SUCCESS";
 // account in case of success or error, if leaveUnlock is not informed.
 export const unlockAcctAndExecFn = (
   passphrase,
-  acctNumber,
+  accts,
   fn,
   leaveUnlock
 ) => async (dispatch, getState) => {
   let res = null;
   let fnError = null;
   dispatch({ type: UNLOCKACCOUNT_ATTEMPT });
+  const walletService = sel.walletService(getState());
 
   // sanity checks
   const accounts = sel.balances(getState());
+  // do not allow locking of the dex account, as it isn't supposed to lock.
+  const dexAccountName = sel.dexAccount(getState());
+  const dexAccount = accounts.find(
+    (acct) => acct.accountName === dexAccountName
+  );
+  accts.forEach((acctNum) => {
+    const account = accounts.find((acct) => acct.accountNumber === acctNum);
+    if (!account) {
+      throw "Account not found";
+    }
+    if (!account.encrypted) {
+      throw "Account not encrypted";
+    }
+  });
 
-  const account = accounts.find((acct) => acct.accountNumber === acctNumber);
-  if (!account) {
-    throw "Account not found";
+  // unlock wallet
+  try {
+    await Promise.all(
+      accts.map(async (acctNumber) => {
+        await wallet.unlockAccount(walletService, passphrase, acctNumber);
+      })
+    );
+    dispatch({ type: UNLOCKACCOUNT_SUCCESS });
+  } catch (error) {
+    await Promise.all(
+      // Need to try and lock all since 1 may have unlocked but not another?
+      accts.map(async (acctNumber) => {
+        if (dexAccount && acctNumber !== dexAccount.accountNumber) {
```

Comment:
This is failing if the wallet doesn't have DEX setup. Needs to be `!dexAccount || acctNumber !== dexAccount.accountNumber`.

Reply:
doh! good catch